lessons/frameworks.ipynb

```diff
@@ -0,0 +1,1261 @@
+    "\n",
+    "* In frequentism, the data are considered to be random variables (in large sets of hypothetical trials described with probability distributions) while parameters are considered fixed (and to be estimated)\n",
+    "\n",
+    "* In Bayesianism, the data are considered to be fixed (as constants, in datafiles) while parameters are considered random variables (to be inferred, with uncertainty described by probability distributions)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Frequentism and Bayesianism\n",
+    "\n",
+    "Given an assumed model:\n",
+    "\n",
+    "* Frequentists seek to _transform_ the frequency distribution of the data into a frequency distribution of their estimators, and hence quantify their uncertainty in terms of _what they expect would happen if the observation were to be repeated_  \n",
+    "  \n",
+    "* Bayesians seek to _update their beliefs_ about their model parameters, and hence quantify their uncertainty in terms of _what might have been had the observation been different_, and _what they knew before the data were taken_\n",
+    "\n",
+    "#### Q: Which approach is better matched to science?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Frequentism vs. Bayesianism\n",
+    "\n",
+    "\n",
+    "* \"You have to assume a prior\" cf. \"You get to assume a prior\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* \"Your calculations are computationally expensive\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* \"How do you account for your nuisance parameters?\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* \"Your conclusions are not relevant\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Things to remember\n",
+    "\n",
+    "* The most important thing is to _know what you are doing_, and to _communicate that clearly to others_\n",
+    "\n",
+    "* Both approaches involve generative models, and their assumptions which must be recorded and tested\n",
+    "\n",
+    "* The Bayesian approach provides a logical framework for combining datasets and additional information, and provides answers in terms of the probability distribution for the model parameters\n",
+    "\n",
+    "* The Frequentist approach provides a way of studying the model independent of additional information, and provides answers in terms of the probability of getting the data"
```

Comment:
I wondered a little bit if frequentism and ML have been conflated confusingly at this point, but maybe it's ok.

Reply:
Dropped the generative model part, because I don't think its necessarily true. I was aiming for more of a "you can't do inference without making assumptions" sense.